autoload/airline/extensions/lsp.vim

```diff
@@ -61,7 +61,44 @@ function! airline#extensions#lsp#get_error() abort
   return airline#extensions#lsp#get('error')
 endfunction
 
+let s:lsp_progress = []
+function! airline#extensions#lsp#progress() abort
+  if get(w:, 'airline_active', 0)
+    if exists('*lsp#get_progress')
+      let s:lsp_progress = lsp#get_progress()
+
+      if len(s:lsp_progress) == 0 | return '' | endif
+
+      " show only most new progress
+      let s:lsp_progress = s:lsp_progress[0]
+      if s:lsp_progress['message'] != '' && s:lsp_progress['percentage'] != 100
+        let percent = ''
+        if s:lsp_progress['percentage'] >= 0
+          let percent = ' ' . string(s:lsp_progress['percentage']) . '%'
+        endif
+        let s:title = s:lsp_progress['title']
+        let message = airline#util#shorten(s:lsp_progress['message'] . percent, 91, 9)
+        return s:lsp_progress['server'] . ':' . s:title . ' ' . message
+      endif
+      return ''
+    endif
+  endif
+endfunction
+
+let s:timer = 0
+let s:ignore_time = 0
+function! airline#extensions#lsp#update() abort
+  if reltimefloat(reltime()) - s:ignore_time >=
```

Comment:
I wonder, if this works with older Vims.

Reply:
vim-lsp also uses reltimefloat() and reltime().
Therefore, I don't think it is necessary to work with older versions of vim any more.